gcp-observability/src/main/java/io/grpc/gcp/observability/interceptors/ConfigFilterHelper.java

```diff
@@ -69,82 +46,41 @@ public class ConfigFilterHelper {
-    List<LogFilter> logFilters = config.getLogFilters();
-    if (logFilters == null) {
-      return;
-    }
 
-    Map<String, FilterParams> perServiceFilters = new HashMap<>();
-    Map<String, FilterParams> perMethodFilters = new HashMap<>();
-
-    for (LogFilter currentFilter : logFilters) {
-      // '*' for global, 'service/*' for service glob, or 'service/method' for fully qualified
-      String methodOrServicePattern = currentFilter.pattern;
-      int currentHeaderBytes
-          = currentFilter.headerBytes != null ? currentFilter.headerBytes : 0;
-      int currentMessageBytes
-          = currentFilter.messageBytes != null ? currentFilter.messageBytes : 0;
-      if (methodOrServicePattern.equals("*")) {
-        // parse config for global, e.g. "*"
-        if (perServiceFilters.containsKey(globalPattern)) {
-          logger.log(Level.WARNING, "Duplicate entry : {0}", methodOrServicePattern);
-          continue;
-        }
-        FilterParams params = FilterParams.create(true,
-            currentHeaderBytes, currentMessageBytes);
-        perServiceFilters.put(globalPattern, params);
-      } else if (methodOrServicePattern.endsWith("/*")) {
-        // TODO(DNVindhya): check if service name is a valid string for a service name
-        // parse config for a service, e.g. "service/*"
-        String service = MethodDescriptor.extractFullServiceName(methodOrServicePattern);
-        if (perServiceFilters.containsKey(service)) {
-          logger.log(Level.WARNING, "Duplicate entry : {0)", methodOrServicePattern);
-          continue;
-        }
-        FilterParams params = FilterParams.create(true,
-            currentHeaderBytes, currentMessageBytes);
-        perServiceFilters.put(service, params);
-      } else {
-        // TODO(DNVVindhya): check if methodOrServicePattern is a valid full qualified method name
-        // parse pattern for a fully qualified method, e.g "service/method"
-        if (perMethodFilters.containsKey(methodOrServicePattern)) {
-          logger.log(Level.WARNING, "Duplicate entry : {0}", methodOrServicePattern);
-          continue;
+  /**
+   * Checks if the corresponding service/method passed needs to be logged according to user provided
+   * observability configuration.
+   * Filters are evaluated in text order, first match is used.
+   *
+   * @param fullMethodName the fully qualified name of the method
+   * @return FilterParams object 1. specifies if the corresponding method needs to be logged
+   *     (log field will be set to true) 2. values of payload limits retrieved from configuration
+   */
+  public FilterParams logRpcMethod(String fullMethodName, Boolean client) {
+    FilterParams params = NO_FILTER_PARAMS;
+
+    int index = checkNotNull(fullMethodName, "fullMethodName").lastIndexOf('/');
+    String serviceName = fullMethodName.substring(0, index);
+    List<LogFilter> logFilters =
+        client ? config.getClientLogFilters() : config.getServerLogFilters();
```

Comment:
This can throw a NullPointerException because `client` is a `Boolean`. It should really be a non-boxed `boolean`.

Reply:
I will update.